srunner/scenarios/background_activity.py

```diff
@@ -1816,16 +1818,25 @@ def _leave_space_in_front(self, space):
 
         self._move_actors_forward(front_actors, space)
 
-    def _remove_lane(self, lane_id):
-        """Remove BA actors from this lane, and BA would never generate new actors on this lane"""
+    def _switch_lane(self, lane_id, active):
+        """
+        active is False: remove BA actors from this lane, and BA would never generate new actors on this lane
+        active is True: recover the lane, reset source waypoint.
+        """
         lane_id = str(lane_id)
         lane_id_list = [x.split('*')[-1] for x in  list(self._road_dict.keys())]
         if lane_id in lane_id_list:
             lane_index = lane_id_list.index(lane_id)
             lane_key = list(self._road_dict.keys())[lane_index]
-            for actor in list(self._road_dict[lane_key].actors):
-                self._destroy_actor(actor)
-            self._road_dict.pop(lane_key)
+            self._road_dict[lane_key].active = active
+            if active:
```

Comment:
This the previous change, nothing should be done when activating the lane source.

Reply:
removed